python/src/array.cpp

```diff
@@ -168,52 +193,91 @@ void get_shape(T list, std::vector<int>& shape) {
       return get_shape(l.template cast<py::list>(), shape);
     } else if (py::isinstance<py::tuple>(l)) {
       return get_shape(l.template cast<py::tuple>(), shape);
+    } else if (py::isinstance<array>(l)) {
+      auto arr = py::cast<array>(l);
+      shape.insert(shape.end(), arr.shape().begin(), arr.shape().end());
```

Comment:
nit:  I think `shape = arr.shape()` is sufficient here.

Reply:
@awni no...
if `a, b = mx.array([1]), mx.array([2])`
then in evaluating `mx.array([a, b])` on line 198 with `shape = arr.shape()` we'd end up with shape = [1], but actually the shape should be [2, 1]. `shape = arr.shape()` will drop a dimension in the process of list unrolling at line 189. You will see what I mean if you see the full block of the `get_shape()`
https://github.com/ml-explore/mlx/blob/1bcee2074324b6ec90d247cf4f118111af5d03bb/python/src/array.cpp#L188-L202

Reply:
Sorry I misunderstood this code, thanks for clarifying.